source/code/include/playfab/PFTitlePlayer.h.ejs

```diff
@@ -0,0 +1,142 @@
+/// are no longer needed.
+/// </remarks>
+HRESULT <%- prefix %>TitlePlayerDuplicateHandle(
+    _In_ <%- prefix %>TitlePlayerHandle titlePlayerHandle,
+    _Out_ <%- prefix %>TitlePlayerHandle* duplicatedHandle
+) noexcept;
+
+/// <summary>
+/// Closes a <%- prefix %>TitlePlayerHandle.
+/// </summary>
+/// <param name="titlePlayerHandle">TitlePlayer handle to close.</param>
+/// <returns>Result code for this API operation.</returns>
+void <%- prefix %>TitlePlayerCloseHandle(
+    _In_ <%- prefix %>TitlePlayerHandle titlePlayerHandle
+) noexcept;
+
+/// <summary>
+/// Gets a generic <%- prefix %>EntityHandle for a TitlePlayer Entity. The <%- prefix %>EntityHandle can be used to
+/// make API calls which only require a generic Entity rather than specifically a TitlePlayer Entity.
+/// </summary>
+/// <param name="titlePlayerHandle"><%- prefix %>TitlePlayerHandle returned from a login call.</param>
+/// <param name="entityHandle">Generic <%- prefix %>EntityHandle to the TitlePlayer Entity.</param>
+/// <returns>Result code for this API operation.</returns>
+/// <remarks>
+/// The output entityHandle and the titlePlayerHandle are separate and both need to be closed when they are no longer needed.
+/// </remarks>
+HRESULT <%- prefix %>TitlePlayerGetEntityHandle(
+    _In_ <%- prefix %>TitlePlayerHandle titlePlayerHandle,
+    _Out_ <%- prefix %>EntityHandle* entityHandle
+) noexcept;
+
+/// <summary>
+/// Get the PlayFabId (master_player_account Id) for the TitlePlayer.
+/// </summary>
+/// <param name="titlePlayerHandle"><%- prefix %>TitlePlayerHandle returned from a login call.</param>
+/// <param name="playFabId">Returned pointer to the playFabId. Valid until the TitlePlayer object is cleaned up.</param>
+/// <returns>Result code for this API operation.</returns>
+HRESULT <%- prefix %>TitlePlayerGetPlayFabId(
+    _In_ <%- prefix %>TitlePlayerHandle titlePlayerHandle,
+    _Outptr_ const char** playFabId
+) noexcept;
+
+/// <summary>
+/// Get the Entity Id for the TitlePlayer.
+/// </summary>
+/// <param name="titlePlayerHandle"><%- prefix %>TitlePlayerHandle returned from a login call.</param>
+/// <param name="entityId">Returned pointer to the entityId. Valid until the TitlePlayer object is cleaned up.</param>
+/// <returns>Result code for this API operation.</returns>
+HRESULT <%- prefix %>TitlePlayerGetEntityId(
+    _In_ <%- prefix %>TitlePlayerHandle titlePlayerHandle,
+    _Outptr_ const char** entityId
+) noexcept;
+
+/// <summary>
+/// Get the cached Entity token.
+/// </summary>
+/// <param name="titlePlayerHandle"><%- prefix %>TitlePlayerHandle returned from a login call.</param>
+/// <param name="entityToken">Returned pointer to the entityToken. The pointer is valid until the TitlePlayer object is cleaned up, though
+/// the token may expire before then.</param>
+/// <returns>Result code for this API operation.</returns>
+HRESULT <%- prefix %>TitlePlayerGetCachedEntityToken(

```

Comment:
I could add it. Are there scenarios where titles would need the SessionTicket?  The main reason I exposed the EntityToken is because PF Lobby needed it.

Reply:
One use case I'm aware of is the client passing the session ticket to a game server and then the server calls https://docs.microsoft.com/en-us/rest/api/playfab/server/authentication/authenticate-session-ticket?view=playfab-rest

Reply:
Ok I will add it.